utils/README.md

```diff
@@ -0,0 +1,406 @@
+# P4 Utils
+
+This folder contains various scripts used to run the code.
+
+## Table of contents
+- [P4 Utils](#p4-utils)
+  - [Table of contents](#table-of-contents)
+  - [Mininet](#mininet)
+  - [Python-Mininet](#python-mininet)
+  - [Mininet and P4](#mininet-and-p4)
+  - [BMv2 Architecture](#bmv2-architecture)
+  - [Simulation Files](#simulation-files)
+    - [P4 File](#p4-file)
+    - [The Control-Plane file](#the-control-plane-file)
+    - [Topology File](#topology-file)
+    - [Mininet Python Script File](#mininet-python-script-file)
+      - [Exercise Instantiation](#exercise-instantiation)
+      - [Running the Exercise](#running-the-exercise)
+      - [Creating the Mininet Network](#creating-the-mininet-network)
+      - [Starting the Mininet Network](#starting-the-mininet-network)
+      - [Programming the Hosts](#programming-the-hosts)
+      - [Programming the Switches](#programming-the-switches)
+      - [Instantiating the Mininet CLI](#instantiating-the-mininet-cli)
+      - [Stopping the Mininet Network](#stopping-the-mininet-network)
+      - [Other Notes](#other-notes)
+        - [Switch Abstraction](#switch-abstraction)
+        - [Default Switch](#default-switch)
+    - [Makefile](#makefile)
+  - [Wrap-up](#wrap-up)
+
+
+ ## Mininet
+Originally created by Bob Lantz, [Mininet](https://github.com/mininet/mininetmininet) is a very complete network emulation tool. Using Mininet, a user can quickly deploy a full network in minutes. As mentioned in [Oliveira et al.](https://ieeexplore.ieee.org/document/6860404), "the possibility of sharing results and tools at zero cost are positive factors that help scientists to boost their researches despite the limitations of the tool in relation to the performance fidelity between the simulated and the real environment."
```

Comment:
Should the URL here be https://github.com/mininet/mininet instead?  The one in this PR leads to a 404